tests/test_quixstreams/test_platforms/test_quix/fixtures.py

```diff
@@ -43,8 +43,11 @@ def mock_quix_portal_api(api_responses: dict = None):
             api_responses = {}
         api_obj = create_autospec(QuixPortalApiService)
         api_obj.default_workspace_id = None
-        for call, data in api_responses.items():
-            api_obj.__getattribute__(call).return_value = data
+        for call, result in api_responses.items():
+            if isinstance(result, Exception):
+                api_obj.__getattribute__(call).side_effect = result
+            else:
+                api_obj.__getattribute__(call).return_value = result
```

Comment:
Let's avoid doing the `__getattribute__` magic inside the factory and simply mock the methods.

It doesn't really make it easier to use but requires you to also know how this factory is implemented.

This is easier to digest and use IMO
```
portal_api = MagicMock(spec_set=QuixPortalAPIService)
portal_api.get_topic.return_value = ...
portal_api.get_topic.side_effect = ...
```